### ISSUE TITLE
bug(ivy): i18n problem with select ICU expressions

### ISSUE THREAD

**Issue by osnoser1** [angular/angular#34018]
# 🐞 bug report

### Affected Package

 The issue is caused by package @angular/....

### Is this a regression?

Yes, works with VE.

### Description

When using a select icu expression throws an exception at runtime.

## 🔬 Minimal Reproduction

Use select ICU expressions in angular apps.
Minimal repo: https://github.com/osnoser1/angular9-i18n-ivy-ssr

* Install dependencies.
* Execute `ng serve -c es`

## 🔥 Exception or Error
<pre><code>
core.js:17 ERROR Error: ASSERTION ERROR: Missing key "other" in ICU statement.
    at throwError (core.js:17)
    at assertGreaterThan (core.js:17)
    at parseICUBlock (vendor.js:49378)
    at extractParts (vendor.js:49276)
    at i18nStartFirstPass (vendor.js:49726)
    at ɵɵi18nStart (vendor.js:49618)
    at Module.ɵɵi18n (vendor.js:50631)
    at AppComponent_Template (app.component.html:305)
    at executeTemplate (core.js:17)
    at renderView (core.js:17)
</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>
9.0.0-rc.3
</code></pre>

**Anything else relevant?**

**Comment by AndrewKushnir** [comment]
Hi @osnoser1, thanks for reporting the issue.

The problem appeared because [an ICU defined in the `app.component.html` template](https://github.com/osnoser1/angular9-i18n-ivy-ssr/blob/master/src/app/app.component.html#L305) has no "other" case, i.e.:

```
{test, select, male {male} female {female}}
```

There is a [check that verifies "other" case presence](https://github.com/angular/angular/blob/master/packages/core/src/render3/i18n.ts#L193) and throws the error. That check was implemented to better align with [ICU format spec](http://site.icu-project.org/design/formatting/select), which states that the "other" case is required:

```
The keyword "other" must occur in the pattern string, or the error U_DEFAULT_KEYWORD_MISSING is returned.
```

However since "other" cases were not required in View Engine, we decided to change this condition in Ivy as well to better align with existing behavior and I've [created a PR to remove the mentioned check](https://github.com/angular/angular/pull/34042). Meanwhile you can update ICUs to add "other" cases (with empty content) to avoid the problem until the mentioned PR lands. Here is an example of the original ICU with the "other" case:

```
{test, select, male {male} female {female} other {}}
```

Thank you.

**Comment by osnoser1** [comment]
Thanks @AndrewKushnir for the detailed explanation and for workaround while merging the PR. Really grateful.

**Comment by AndrewKushnir** [comment]
@osnoser1, just want to let you know that the fix from the mentioned PR was included into v9.0.0-rc.4, which was released a few hours ago and is now available on NPM. I'll close this ticket, but feel free to reopen it if the problem still exists. Thank you.

**Comment by osnoser1** [comment]
Thank you very much @AndrewKushnir 😄
